src/routers/alpha-router/functions/get-candidate-pools.ts

```diff
@@ -357,149 +278,265 @@ export async function getV3CandidatePools({
     );
   }
 
-  addToAddressSet(top2DirectSwapPool);
+  addToAddressSet(topByDirectSwapPools);
 
+  const baseTokens = baseTokensByChain[chainId] ?? [];
+  const baseTokensAddresses = baseTokens.map((token) => token.address.toLowerCase());
```

Comment:
nit: I think it will be _slightly_ quicker if we do lower casing in the if block within the for loop, since topNWithEachBaseToken < baseTokensByChain[mainnet], and topNWithBaseToken < baseTokensByChain[mainnet]

might not impact perf that much, if lower casing is a cheap operation.

Reply:
I don't think thats necessarily true, it depends of how many iterations we do before we fill `topNWithBaseToken` that could happen in 4 iterations or 1000.

`baseTokenByChain` is usually a super small list, manually curated